base/multidimensional.jl

```diff
@@ -579,7 +579,7 @@ end
 
         storeind = 1
         Xc, Bc = X.chunks, B.chunks
-        idxlens = index_lengths(B, I0, I...) # TODO: unsplat?
+        idxlens = @ncall $N index_lengths B I0 d->I[d]
```

Comment:
I'm pretty sure the splat here is fine — `index_lengths` always inlines so it should be equivalent to the `@ncall` version.  But it's definitely easier to reason about the performance of `@ncall`, so this is fine.